### DISCUSSION TITLE
Tutorial: Flutter > Firebase (business logic) > Hasura GraphQL > PostgreSQL - not working

### DISCUSSION THREAD

**Discussion by well-i-never** [hasura/graphql-engine#10616] [Question] [answered]
Hi,

I am trying to follow the tutorial: https://hasura.io/blog/build-flutter-app-graphql-hasura-serverless-part1.
Unfortunately I am stuck on part 2. The code to set up on index.js for the firebase project does not compile.

![image](https://github.com/user-attachments/assets/1ce4f740-249d-41bd-b21c-8f02c7147756)

Also, the tutorial is missing some bits. When "firebase init" is called, there are questions to the user which are not addressed in the tutorial.

Also it would be great to understand how to set up authentication directly in Hasura as opposed to using Heroku. (I am unclear why Heroku is used, but this could be due to my lack of understanding. I have used render.com to host the postgreSQL database for the purpose of trying to get this tutorial working as it is free).

Any guidance would be greatly appreciated.

**Comment by robertjdominguez** [comment, marked as answer]
Hi, @well-i-never — sorry for the delay in a response here!

FIrst, this blog post is _severely_ out of date 😅 

As you can see, it was authored in 2019. FWIW, the error you're seeing seems to be related to how node handles ESM vs. common JS when it comes to exports. You can read more about that [here](https://blog.logrocket.com/commonjs-vs-es-modules-node-js/). Don't worry — it confuses us all!

Regardless, I'd like to focus on your second point:

> Also it would be great to understand how to set up authentication directly in Hasura as opposed to using Heroku. (I am unclear why Heroku is used, but this could be due to my lack of understanding. I have used render.com to host the postgreSQL database for the purpose of trying to get this tutorial working as it is free).

When this was written, Heroku was used as our preferred deployment platform; additionally, we used their now-deprecated free tier to serve PostgreSQL instances for Hasura Cloud projects. However, that's not the case anymore.

Hasura will work with nearly _any_ authentication provider using webhooks or JWTs. You can read more about them [here](https://hasura.io/docs/2.0/auth/overview/) or try a quickstart using [this link](https://hasura.io/docs/2.0/auth/quickstart/).

Do let us know if you have other questions!
